harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -1591,3 +1583,188 @@ def test_vm_with_bogus_vlan(api_client, image, unique_vm_name,
         f"Failed to delete VM({unique_vm_name})'s volumes with errors:\n"
         "\n".join(f"Volume({n}): {r}" for n, r in fails)
     )
+
+
+@pytest.mark.p0
+@pytest.mark.virtualmachines
+class TestHotPlugVolume:
+    """
+    To cover test:
+    - https://harvester.github.io/tests/manual/volumes/support-volume-hot-unplug/
+
+    Steps:
+        1. Create and start VM
+        2. Create Data volume
+        3. Attach data volume
+        4. Detach data volume
+    Exepected Result:
+        - VM should started successfully
+        - Data volume should attached and available in VM
+        - Data volume should detached and unavailable in VM
+        - VM should not be reboot or restart while attaching/detaching volume
+    """
+
+    disk_name = "disk-hot-plug"
+
+    @pytest.mark.dependency(name="hot_plug_volume")
+    def test_add(
+        self, api_client, ssh_keypair, wait_timeout, host_shell, vm_shell, small_volume, stopped_vm
+    ):
+        unique_vm_name, ssh_user = stopped_vm
+        pub_key, pri_key = ssh_keypair
+
+        # Start VM
+        code, data = api_client.vms.start(unique_vm_name)
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.vms.get_status(unique_vm_name)
+            if 200 == code:
+                phase = data.get('status', {}).get('phase')
+                conds = data.get('status', {}).get('conditions', [{}])
+                if ("Running" == phase
+                   and "AgentConnected" == conds[-1].get('type')
+                   and data['status'].get('interfaces')):
+                    break
+            sleep(3)
+        else:
+            raise AssertionError(
+                f"Failed to Start VM({unique_vm_name}) with errors:\n"
+                f"Status: {data.get('status')}\n"
+                f"API Status({code}): {data}"
+            )
```

Comment:
I feel, we should have a fixture `running_vm` instead of writing `starting vm` code in the test. As in future, many test cases will require running vm to start with.
Could you help to implement that?

Reply:
we need to start the VM because we leverage existing `stopped_vm` which already defined, stopped_vm would be more useful when we are going to update the VM or especially **cloud-config**.

Of course we can add another `running_vm` fixture, but as you mentioned in next comment, it will add more lines rather than check the VM is started.

It is a trade off, in here, I would prefer reuse existing fixture and start the VM to make sure the VM just started and accessible.

Reply:
1. Can you elaborate on how the `running_vm` will add more lines? I'm assuming a running_vm fixture will have the checks that the vm is started successfully.

2. Running_vm could be useful, quick and convenient to use for other test cases.

Reply:
2 lines for function naming and decorator, 1 lines for return, and lines same as the codes, is it the minimal lines to create the fixture and depends on stopped_vm